Fixes armcannon messages displaying to everybody

PKPenguin321 opened the pull request:
Fixes https://github.com/tgstation/-tg-station/issues/14712

This was tested, so it should be good

PKPenguin321 committed:
fixes armcannon messages

they were displaying to everybody instead of just the user because i misunderstood how messages work.

phil235 commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -291,7 +291,7 @@
 		holder.loc = null
 		out = 0
-		owner.visible_message("<span class='notice'>[holder] forcibly retracts into your arm.</span>")
+		owner << "<span class='warning'>[holder] forcibly retracts into your arm.</span>"
```
This means the owner will get the message even if they're dead when the implant gets EMPed.

duncathan replied:
Is that an issue?

phil235 replied:
Sort of. It's not a big deal though. It's similar to getting "you feel woozy" while dead and low on health, not very realistic.

phil235 commented:
Why not simply add a second argument to visible_message() so you have one message for the owner and another for other viewers?

PKPenguin321 commented:
Viewers don't really need a message, they're just feedback for the user to confirm that it's working as intended

Dorsisdwarf commented:
<< isnt the best here, I think

PKPenguin321 commented:
It works, and that's really all I care about, but if you have a better alternative hit me up

PKPenguin321 commented:
So does anybody have an alternative or can this be merged

PKPenguin321 committed:
adds a safety check to prevent runtimes

if the dummy parent item somehow gets spawned and an idiot player tries to use it, this stops them from runtiming/crashing the server. written in my free time so i may have done the "holder == null" part wrong, if it breaks ill fix it in a bit.

danhibiki commented:
I'm pretty sure if you just do visible_message instead of owner.visible_message, it will only display to the parent mob with one argument. I did something similar on a different codebase relating to a weapon, when before I changed it it had the same issue you are having with this, and it was calling owner.visible_message as well.

phil235 commented:
@danhibiki No, visible_message() always display messages to all viewers.
